Make `ButtonProps.onClick` more type-able.

johnwchadwick opened the pull request:
This PR swaps the order of the `onClick` arguments so that the optional `value` argument comes second instead of first. It's difficult to statically verify that this PR is correct, especially because we do not have `noImplicitAny` enabled. However, `onClick` is fairly tricky to type correctly as-is, which is a barrier to getting to `noImplicitAny` right now. I've attempted to manually trace all references to `Button` across components to find ones which simultaneously use `onClick` and `value` props, as those are the only ones that are impacted.

johnwchadwick committed:
Make Button onClick prop more typeable

Swaps the order of the arguments so that the optional arg is second.

jackkav commented on packages/insomnia/src/ui/components/modals/request-switcher-modal.tsx:
```diff
@@ -485,7 +489,7 @@ class RequestSwitcherModal extends PureComponent<Props, State> {
                 return (
                   <li key={w._id}>
-                    <Button onClick={this._activateWorkspace} value={w} className={buttonClasses}>
+                    <Button onClick={(_e, w) => this._activateWorkspace(w)} value={w} className={buttonClasses}>
```
This appears to be defining `w` inside the onClick when it's already defined in outer scope at the `matchedWorkspaces.map`.

johnwchadwick replied:
Fixed.

marckong commented on packages/insomnia/src/ui/components/base/button.tsx:
```diff
@@ -7,8 +7,8 @@ export interface ButtonProps<T> {
   className?: string;
-  onDisabledClick?: React.MouseEventHandler<HTMLButtonElement> | ((value: T | undefined, event: React.MouseEvent<HTMLButtonElement>) => void);
-  onClick?: React.MouseEventHandler<HTMLButtonElement> | ((value: T | undefined, event: React.MouseEvent<HTMLButtonElement>) => void);
+  onDisabledClick?: (event: React.MouseEvent<HTMLButtonElement>, value: T) => void;
```
(value: T) could be undefined from the existing the type. I wonder if making it required is intentional.

johnwchadwick replied:
Truth be told, it's actually inconsequential since T is only ever not provided when T = undefined as far as I know. I should probably just make T "required" on both and simplify the code.

johnwchadwick replied:
Fixed. I did not attempt to simplify it yet, in interest of trying to keep the code the same.

marckong commented on packages/insomnia/src/ui/components/editors/auth/components/auth-toggle-row.tsx:
```diff
@@ -27,7 +27,7 @@ export const AuthToggleRow: FC<Props> = ({
 
   const databaseValue = Boolean(authentication[property]);
-  const toggle = useCallback((value: boolean) => patchAuth({ [property]: value }), [patchAuth, property]);
+  const toggle = useCallback((_event, value: boolean) => patchAuth({ [property]: value }), [patchAuth, property]);
```
was the type for _event skipped intentionally due to its being unused?

johnwchadwick replied:
Probably, but we can just specify it. (Fixed.)

marckong commented on packages/insomnia/src/ui/components/modals/request-switcher-modal.tsx:
```diff
@@ -238,7 +242,7 @@ class RequestSwitcherModal extends PureComponent<Props, State> {
     const { workspace, workspaceRequestsAndRequestGroups, workspacesForActiveProject, requestMetas, grpcRequestMetas, activeRequest } = this.props;
     const { maxRequests, maxWorkspaces, hideNeverActiveRequests } = this.state;
-    const lastActiveMap = {};
+    const lastActiveMap: {[key: string]: number} = {};
```
I am wondering if there was any reason using this instead of Record<string, number>.

johnwchadwick replied:
Nope, force of habit. I've been using TypeScript for a decent bit longer than the `Record` alias has existed.

johnwchadwick replied:
Fixed.

johnwchadwick committed:
Review fixes.

dimitropoulos committed:
removes some implicit any's

dimitropoulos committed:
NoticeTable to FC

might as well, at this point.  we're trying to do these as we can as they present themselves

jackkav commented on packages/insomnia/src/ui/components/modals/request-switcher-modal.tsx:
```diff
@@ -452,7 +456,7 @@ class RequestSwitcherModal extends PureComponent<Props, State> {
                 return (
                   <li key={r._id}>
-                    <Button onClick={this._activateRequest} value={r} className={buttonClasses}>
+                    <Button onClick={(_e, r) => this._activateRequest(r)} value={r} className={buttonClasses}>
```
also ambiguous `r`

dimitropoulos replied:
ambiguous how?  you mean because of shadowing?

jackkav replied:
As in my previous comment. 
It's instantiated in both parent and child scope with the same name.

johnwchadwick replied:
To be fair, it's not actually ambiguous. Shadowing behavior is exact.

dimitropoulos replied:
my thoughts exactly: I'm confused what is ambiguous, still

jackkav replied:
This is now inconsistent https://github.com/Kong/insomnia/pull/4853#discussion_r894411327

johnwchadwick replied:
Ah, I did mean to still address this before merging. However, the code behaves correctly: the shadowed `r` is still going to take precedence. This is merely just a readability issue. So it's not a huge concern. I can fix it in a follow-up if you'd like.

dimitropoulos committed:
fixes margin for collapse icon (broken before this PR)

and removes an unnecessary type parameter

dimitropoulos reviewed (approved):
this change kinda scares me because I thought it was so much more ubiquitous than it tuned out to be.

I went through every match for `<Button` in the app (case sensitive) and checked them all and found a few very minor fixes in 6fe19c35365fb8ccb02e1aa3bb1f77df269ed95a.

we're trying to do FC conversions as they present themselves for little 1-method classes, so I did so in d95711d20a4570a81087079f2cf2b4e1d9760839 (but lemme know if you want me to pull out into separate PR).

also I noticed the margin was broken on the button in this component, so I fixed it in d360346373e40c1efdc5ea0e79b648dbb1635126

| BEFORE | AFTER |
| - | - |
| ![Screenshot_20220610_152006](https://user-images.githubusercontent.com/15232461/173135498-a58b72c1-72a5-4958-b7d1-c7dae0e95a7c.png) | ![Screenshot_20220610_151942](https://user-images.githubusercontent.com/15232461/173135457-a0fc3483-e708-472e-b1c8-7f0dc77f1d42.png) |
